File: src/main/java/info/skyblond/archivedag/config/CertSigningProperties.java
package info.skyblond.archivedag.config;

import lombok.Getter;
import lombok.Setter;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.core.io.Resource;

import java.util.concurrent.TimeUnit;

@Getter
@Setter
@ConfigurationProperties(value = "cert-signing")
public class CertSigningProperties {
    private Resource caCert = null;
    private Resource caPrivateKey = null;
    private String caPrivateKeyPassword = "";
    private String customSubjectDN = "";
    private String signAlgName = "SHA3-512WITHRSA";
    private int generateKeySize = 4096;
    private long expireInDuration = 90;
    private TimeUnit expireInUnit = TimeUnit.DAYS;
}

File: backend/src/main/kotlin/info/skyblond/archivedag/arstue/config/CertSigningProperties.kt
package info.skyblond.archivedag.arstue.config

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.context.properties.ConstructorBinding

@ConstructorBinding
@ConfigurationProperties(value = "cert-signing")
data class CertSigningProperties(
    /**
     * Subject C: Country name
     * */
    val subjectDnC: String? = null,
    /**
     * Subject O: Organization name
     * */
    val subjectDnO: String? = null,
    /**
     * Subject OU: Organization unit name
     * */
    val subjectDnOU: String? = null,
)
